flycheck.el

```diff
@@ -3942,8 +3967,10 @@ Return a list with the contents of the table cell."
                    column 'flycheck-error-list-column-number)
                   (flycheck-error-list-make-cell
                    (symbol-name (flycheck-error-level error)) level-face)
-                  (flycheck-error-list-make-cell
-                   id-str 'flycheck-error-list-id id-str)
+                  ;; Error ID has a button to explain the error
+                  (list id-str
+                        'type 'flycheck-error-list-explain-error
```

Comment:
Checking the explainer is just checking the nullness of the property on the error checker, so no, it shouldn't be costly.  But it seems more confusing to have the button sometimes link to the error (when there is no explainer) and sometimes to the explanation (when there is one).  

Maybe a compromise would be to make the button 'active' somehow (using faces?) when there is an explainer, so the user can know just by looking at the error list which errors can be explained?